### DISCUSSION TITLE
Attempting a git reset or revert I think.

### DISCUSSION THREAD

**Discussion by strikeouts27** [community/community#60452] [Programming Help] [answered]
### Body

I was working on my django project and I encountered a bug. A friend of mine tried to help. Unfortunately we couldn't solve it. I had pushed changes incrementally up to Github. I see my prior commit and I tried to download the zip files from where I wanted to continue. When I tried working on the code again, it synchronized back to the most current commit. Strange I did not know it could do that. 

I need to go back to my part 4 commit 

My repo 
https://github.com/strikeouts27/mysite

Any advice?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by GhostOf0days** [comment, marked as answer]
Make sure you backup repo if you need the new commits.

```git reset --hard HEAD~x```, where x is a number representing how many commits you want to go back. x=1 means go back to the commit before the most recent.

or 

```git reset --hard commit-id```, where commit-id is a string representing the commit you want to reset to. This can be found in your repo's commit history.

Then do ```git push --force origin branch-name```, where branch-name is main if you would like to reset the branch where you should be making new branches off of with ```git checkout -b branch-name``` after ```git checkout main```.

**Reply by GhostOf0days** [reply]
Get your commit ID from [here](https://github.com/strikeouts27/mysite/commits/main). I won't type it here in case you decide to make the repo private or delete it later on. The commit message is 'part 4 of tutorial complete'. The commit ID is the string on the right side that you are able to copy.

**Reply by strikeouts27** [reply]
I have made a backup repository. And I have run the git reset --hard commit-id command and now I am back to what it was before.

I want my repository to operate from this point in time forward and have it be the main branch. 

are you telling me to use git checkout once I have run the git push command and make a new branch and have it run from that point forward?

**Reply by GhostOf0days** [reply]
Just do it on the main repo. Run ```git reset --hard commit-id``` then ```git push --force origin main``` on your main repo. Now, your main repo will be at the point you want.

**Reply by strikeouts27** [reply]
Thank you GhostOf0days! :) Really. Thank you! :)

**Reply by GhostOf0days** [reply]
No problem. Happy to help. 🙂

**Reply by strikeouts27** [reply]
one moment my repository is not showing the git push you wanted me to do as the most recent commit. did I need to do a git add . and than a git commit and than push?

**Reply by strikeouts27** [reply]
nevermind it just took a minute to update :)

**Reply by danielecr** [reply]
Thank you, yours is one of the most usefull suggestion on reset --hard, but English is not my first language, at the end I find this statement confusing:

> Then do `git push --force origin branch-name`, where branch-name is main if you would like to reset the branch where you should be making new branches off of with `git checkout -b branch-name` after `git checkout main`

after the first comma, it is hard for me to build a timeline of suggested actions. (really, for me it is like seeing Pulp Fiction the first time 😄 )
